src/replication.c

```diff
@@ -190,6 +198,77 @@ void rebaseReplicationBuffer(long long base_repl_offset) {
     }
 }
 
+/* Replication: Primary side - connections association.
+ * On rdb-connection sync, connection association is used to keep replication data in
+ * the backlog until the replica requests PSYNC. Association happens in two forms,
+ * if there's an existing buffer block at the fork time, the replica is attached
+ * to the tail, if there is no tail, the replica will be attached when a new
+ * buffer block is created (see the Retrospect function below).
+ * Replica rdb client id is used as a unique key for the association.
+ * On COB overrun, association is deleted and the RDB connection
+ * is dropped.
+ */
+void addReplicaToPsyncWait(client *replica) {
+    listNode *ln = NULL;
+    replBufBlock *tail = NULL;
+    if (server.repl_backlog == NULL) {
+        createReplicationBacklog();
+    } else {
+        ln = listLast(server.repl_buffer_blocks);
+        tail = ln ? listNodeValue(ln) : NULL;
+        if (tail) {
+            tail->refcount++;
+        }
+    }
+    serverLog(LL_DEBUG, "Add replica %s to waiting psync rax, with cid %llu, %s ", replicationGetReplicaName(replica),
+              (unsigned long long)replica->id, tail ? "with repl-backlog tail" : "repl-backlog is empty");
+    replica->ref_repl_buf_node = tail ? ln : NULL;
+    /* Prevent rdb client from being freed before psync is established. */
+    replica->flags |= CLIENT_PROTECTED_RDB_CONN;
+    uint64_t id = htonu64(replica->id);
+    raxInsert(server.replicas_waiting_psync, (unsigned char *)&id, sizeof(id), replica, NULL);
+}
+
+/* Attach waiting psync replicas with new replication backlog head. */
+void addReplicaToPsyncWaitRetrospect(void) {
```

Comment:
I am not sure about retrospect. It is not very clear about the operation IMO. maybe `backfill`? thoughts?

```suggestion
void backfillRdbReplicasToPsyncWait(void) {
```

Reply:
I agree. Back fill better describes what we do here